src/structs.rs

```diff
@@ -61,132 +61,120 @@ impl Keyword {
     }
 }
 
+#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq)]
+pub enum BlockType {
+    Addition,
+    Deletion,
+}
+
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq)]
+pub enum BlockPrefix {
+    NonTerminating,
+    Terminating { length: u8 },
+    Padding,
+}
+
+impl From<BlockPrefix> for u8 {
+    fn from(value: BlockPrefix) -> Self {
+        match value {
+            BlockPrefix::NonTerminating => u8::MAX,
+            BlockPrefix::Terminating { length } => length,
+            BlockPrefix::Padding => 0,
+        }
+    }
+}
+
+impl From<u8> for BlockPrefix {
+    fn from(value: u8) -> Self {
+        if 0 == value {
+            Self::Padding
+        } else if u8::MAX == value {
+            Self::NonTerminating
+        } else {
+            Self::Terminating { length: value }
+        }
+    }
```

Comment:
Maybe use a match here to have symmetry with the inverse operation above?

Reply:
done in f2c5ec6